overthebox/files/etc/udhcpc.user

```diff
@@ -37,60 +37,46 @@ set dhcpdiscovery.$INTERFACE.serverid=$serverid
+uci -q show network | grep -sq "network\.if.*\.gateway='$router'" && exit
 
-if [ -z "$KNOWN_ROUTER" ]; then
+_id=1
+while uci -q get network.if$_id >/dev/null; do _id=$((_id+1)); done
 
-	if [ "$(uci -q get network.lan.proto)" != "static" ]; then
-		uci -q batch <<-EOF
-		set network.lan.proto=static
-		set network.lan.ipaddr=$ip
-		set network.lan.netmask=$subnet
-		set network.lan_dev.macaddr=$_macaddr
-		set network.if0_dev.macaddr=auto.lan
-		commit network
-		EOF
-	else
-		_id=1
-		while uci -q get network.if$_id > /dev/null; do _id=$((_id+1)); done
+if uci -q get "network.lan_if${_id}" >/dev/null; then
+	uci -q batch <<-EOF
+	set network.if${_id}_dev=device
+	set network.if${_id}_dev.name=if$_id
+	set network.if${_id}_dev.type=macvlan
+	set network.if${_id}_dev.ifname=$_ifname
+	set network.if${_id}_dev.macaddr=$_macaddr
+	set network.if${_id}=interface
+	set network.if${_id}.ifname=if$_id
+	set network.if${_id}.type=macvlan
+	set network.if${_id}.proto=static
+	set network.if${_id}.ipaddr=$ip
+	set network.if${_id}.netmask=$subnet
+	set network.if${_id}.multipath=on
+	set network.if${_id}.gateway=$router
+	set network.if${_id}.dns=$dns
+	set network.if${_id}.metric=$_id
+	set network.if0_dev.macaddr=auto.if$_id
+	commit network
 
-		uci -q batch <<-EOF
-		set network.if${_id}_dev=device
-		set network.if${_id}_dev.name=if$_id
-		set network.if${_id}_dev.type=macvlan
-		set network.if${_id}_dev.ifname=$_ifname
-		set network.if${_id}_dev.macaddr=$_macaddr
-		set network.if${_id}=interface
-		set network.if${_id}.ifname=if$_id
-		set network.if${_id}.type=macvlan
-		set network.if${_id}.proto=static
-		set network.if${_id}.ipaddr=$ip
-		set network.if${_id}.netmask=$subnet
-		set network.if${_id}.multipath=on
-		set network.if${_id}.gateway=$router
-		set network.if${_id}.dns=$dns
-		set network.if${_id}.metric=$_id
-		set network.if0_dev.macaddr=auto.if$_id
-		add_list firewall.mptcp.network=if$_id
-		commit network
-		commit firewall
-		EOF
+	add_list firewall.mptcp.network=if$_id
+	commit firewall
 
-		uci -q batch <<-EOF
-		set dhcp.if${_id}=dhcp
-		set dhcp.if${_id}.interface='if${_id}'
-		set dhcp.if${_id}.start='1'
-		set dhcp.if${_id}.leasetime='12h'
-		set dhcp.if${_id}.dynamicdhcp='0'
-		set dhcp.if${_id}.force='true'
-		set dhcp.if${_id}_gw=tag
-		add_list dhcp.if${_id}_gw.dhcp_option='1,$subnet'
-		add_list dhcp.if${_id}_gw.dhcp_option='3,$router'
-		add_list dhcp.if${_id}_gw.dhcp_option='6,$dns'
-		commit dhcp
-		EOF
-	fi
-
-	/etc/init.d/network reload
-	/etc/init.d/firewall reload
+	set dhcp.if${_id}_gw=tag
+	add_list dhcp.if${_id}_gw.dhcp_option=1,$subnet
+	add_list dhcp.if${_id}_gw.dhcp_option=3,$router
+	add_list dhcp.if${_id}_gw.dhcp_option=6,$dns
+	commit dhcp
+	EOF
+else
+	uci -q batch <<-EOF
+	set network.lan_if${_id}.proto=static
+	set network.lan_if${_id}.ipaddr=$ip
+	set network.lan_if${_id}.netmask=$subnet
+	set network.if0_dev.macaddr=auto.lan
```

Comment:
This does not trigger a dhcp request, it does nothing...
Changing back this line to ```set network.if0_dev.macaddr=auto.if$_id``` seems to work